src/middlewared/middlewared/plugins/update_/download.py

```diff
@@ -38,6 +39,13 @@ def download_impl_scale(self, job, train, location, progress_proportion):
                 self.middleware.logger.warning("Invalid update file checksum %r, re-downloading", checksum)
                 os.unlink(dst)
 
+            st = os.statvfs(os.path.dirname(dst))
+            bsize = st.f_bsize if st.f_bsize == 4096 else 512
+            avail = st.f_bavail * bsize
+            if (1024**3) * 2 > avail:
```

Comment:
Let's add `size` to https://update.freenas.org/scale/TrueNAS-SCALE-Angelfish/manifest.json (it is built by scale-build repo)